dashboard/app/views/layouts/_header.html.haml

```diff
@@ -53,9 +53,15 @@
               = render partial: 'shared/user_header'
 
 - if script_level
-  - user_progress = current_user == nil ? 'null' : summarize_user_progress(script_level.script, @user || current_user).to_json
 
   :javascript
     //<![CDATA[
-    dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{user_progress}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'})
+    var buildHeader = function () {
+        dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'});
+    };
+    if(#{!!(@current_user).to_json}) {
```

Comment:
This won't work for publicly cacheable pages because we need to serve the same page to everyone. 

Let's talk about this.  I am changing the sign-in button to be an Ajax hit, we could set some state at that point to let you know the user is signed in or perhaps even piggyback the progress state into the response from that request.